File: frontend/web/parceldelivery-admin-app/src/app/models/Vehicle.ts
export interface Vehicle {
  id: string;
  registrationNumber: string;
  type: string;
  year: number;
  technicalInspectionExpirationDate: Date;
  seatingCapacity: number;
  maxInternalSpaceX: number;
  maxInternalSpaceY: number;
  maxInternalSpaceZ: number;
}

File: frontend/mobil/parceldelivery_mobile/lib/models/vehicle.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'vehicle.freezed.dart';
part 'vehicle.g.dart';

@freezed
class Vehicle with _$Vehicle {
  const factory Vehicle({
    required String id,
    required String registrationNumber,
    required String type,
    required int year,
    required DateTime technicalInspectionExpirationDate,
    required int seatingCapacity,
    required int maxInternalSpaceX,
    required int maxInternalSpaceY,
    required int maxInternalSpaceZ,
  }) = _Vehicle;

  factory Vehicle.fromJson(Map<String, Object?> json) =>
      _$VehicleFromJson(json);
}
